### ISSUE TITLE
RetroArch menu fails to spawn.

### ISSUE THREAD

**Issue by orbea** [loganmc10/GLupeN64#10]
OS: `Slackware64-current`
WM: `spectrwm-3.0.2-x86_64-1_SBo`
`RetroArch-1.3.6-x86_64-1_SBo`

When loading a game from the command line and then trying to spawn the RetroArch menu it will fail to spawn properly. Closing the menu works as expected, however if I try to take a scrot with a key bind (shell script) RetroArch will lose focus and I will then be unable to regain focus forcing me to kill the process. This script has worked flawlessly with RetroArch in the past.

This will occur with the rgui, glui and xmb menu drivers.

![1](http://ks392457.kimsufi.com/orbea/stuff/pics/scrots/libretro/mupen64plus/2016-07-28-110309_1680x1050_scrot.png)

screenshot.sh
```
#!/bin/sh

screenshot() {
    case $1 in
    full)
        /usr/bin/scrot -m
        ;;
    window)
        sleep 1
        /usr/bin/scrot -s
        ;;
    *)
        ;;
    esac;
}

screenshot $1
```

**Comment by loganmc10** [comment]
I suspected something like this would probably happen, I'll have to fix this tonight when I'm in front of my computer, but I'm almost certain it is because I stripped out most of this function (trying to make it more efficient):

https://github.com/loganmc10/GLupeN64/blob/master/custom/glsm/glsm.c#L2076-L2088

If you make it look like this again:
https://github.com/libretro/libretro-common/blob/1e28482cf3d7a0f06a176fb7efb05462df84fd5c/glsm/glsm.c#L2009-L2066

It will probably work fine, but I'll sort it all out tonight.

**Comment by loganmc10** [comment]
Should be fixed with ebc7721ab83f973ad29faa3fd8f61716a80c8ad6, if the menu and your screenshots work please feel free to close this issue.

**Comment by orbea** [comment]
Thanks, both have been resolved.
